internal/exec/stages/files/filesystemEntries.go

```diff
@@ -315,14 +319,16 @@ func (tmp dirEntry) create(l *log.Logger, u util.Util) error {
 	case os.IsNotExist(err):
 		// use default perms, we'll fix it later
 		if err := os.MkdirAll(d.Path, util.DefaultDirectoryPermissions); err != nil {
-			return fmt.Errorf("Failed to create directory %s: %v", d.Path, err)
+			return fmt.Errorf("failed to create directory %s: %v", d.Path, err)
 		}
 	case err != nil:
 		return fmt.Errorf("stat() failed on %s: %v", d.Path, err)
 	case !st.Mode().IsDir():
 		return fmt.Errorf("error creating directory %s: A non-directory already exists and overwrite is false", d.Path)
 	}
-
+	if d.Mode != nil {
+		d.Mode = cutil.IntToPtr(int(util.ToFileMode(d.Mode)))
```

Comment:
Ah, I phrased https://github.com/coreos/ignition/pull/1325#discussion_r835832582 ambiguously.  I meant that `SetPermissions` should do the translation internally (into a local variable) and `f.Mode`/`d.Mode` shouldn't be changed.